test/test_jit.py

```diff
@@ -3360,8 +3370,17 @@ def func():
             print(b)
         ''')
 
-        # test that equal tuples correctly work with node hashing
-        for tup in map(lambda x: "(" + x + ",)", constants):
+        # generate dicts with built-in types (excluding torch.Tensor)
+        xprod = itertools.product(constants, constants)
+
+        def keys_pred(t):
+            return isinstance(eval(t[0]), (list, bool)) or eval(t[0]) is None
+
+        filt = [x for x in xprod if not keys_pred(x)]
+        dict_strs = map(lambda t: '{' + t[0] + ':' + t[1] + '}', filt)
+
+        # test that equal tuples and dicts correctly work with node hashing
+        for tup in chain(map(lambda x: "(" + x + ",)", constants), dict_strs):
             funcs_str = funcs_template.format(constant_constructor=tup)
             scope = {}
             execWrapper(funcs_str, globals(), scope)
```

Comment:
Do you mind adding a test for both the empty list dictionary case, `Dict[int, int]` and the Element Cannot be inferred case, `Dict[int, Optional[int]`. 

Just have a function:
```
def foo():
     a: Dict[int, int] = {}
     b: Dict[int, Optional[int]] = {}
     return a, b
```
run constant propagation, run export import.